src/main/java/pl/psnc/dei/ui/components/batches/DatasetImportComponent.java

```diff
@@ -0,0 +1,308 @@
+        limitRecordsToRetrieveNumberField.setEnabled(doLimitNumberOfRecordsToRetrieve);
+        limitRecordsToRetrieveCheckbox.addValueChangeListener(event -> {
+            doLimitNumberOfRecordsToRetrieve = event.getValue();
+            limitRecordsToRetrieveNumberField.setEnabled(doLimitNumberOfRecordsToRetrieve);
+        });
+        return checkbox;
+    }
+
+    private Component prepareRecordsSplittingInput() {
+        splitRecordsCheckboxWithNumber = new CheckboxWithNumberInput("Split records into several imports with a maximum size of");
+        splitRecordsCheckboxWithNumber.setEnabled(doCreateImport);
+        splitRecordsCheckbox = splitRecordsCheckboxWithNumber.getCheckbox();
+        splitRecordsNumberField = splitRecordsCheckboxWithNumber.getNumberField();
+        splitRecordsNumberField.setEnabled(doSplitRecords);
+        splitRecordsCheckbox.addValueChangeListener(event -> {
+            doSplitRecords = event.getValue();
+            splitRecordsNumberField.setEnabled(doSplitRecords);
+        });
+        return splitRecordsCheckboxWithNumber;
+    }
+
+    private TextField prepareImportNameInputField() {
+        importNameTextField = new TextField();
+        importNameTextField.setLabel("Import name");
+        importNameTextField.getStyle().set("padding-top", "6px");
+        return importNameTextField;
+    }
+
+    private Button prepareUploadRecordsButton() {
+        Button button = new Button("Upload records");
+        button.addClickListener(event -> uploadRecords());
+        return button;
+    }
+
+    private Checkbox prepareCreateImportCheckbox() {
+        createImportCheckbox = new Checkbox();
+        createImportCheckbox.setLabel("Create import");
+        createImportCheckbox.setValue(doCreateImport);
+        createImportCheckbox.addClassName("mt-16px");
+        createImportCheckbox.addValueChangeListener(event -> {
+            doCreateImport = event.getValue();
+            importNameTextField.setEnabled(doCreateImport);
+            splitRecordsCheckboxWithNumber.setEnabled(doCreateImport);
+        });
+        return createImportCheckbox;
+    }
+
+    private Component createAggregatorSelection() {
+        Predicate<Aggregator> predicate = agg -> agg.equals(Aggregator.EUROPEANA);
+        aggregatorsSelect = CommonComponentsFactory.getAggregatorSelector(predicate);
+        aggregatorsSelect.setEnabled(false);
+        aggregatorsSelect.setValue(Aggregator.EUROPEANA);
+        return aggregatorsSelect;
+    }
+
+    private Component createProjectSelection() {
+        ListDataProvider<Project> projectsProvider = new ListDataProvider<>(projectsRepository.findAll());
+        projectsSelect = CommonComponentsFactory.getProjectSelector(projectsProvider);
+        projectsSelect.addValueChangeListener(event -> {
+            project = event.getValue();
+            datasetsSelect.setItems(batchService.getProjectDataset(project));
+        });
+        return projectsSelect;
```

Comment:
code duplication, with `BatchImportComponent::createProjectSelection`. I thing it is worth to extract it to overloaded method in `CommonComponentsFactory`, as this method will pop-up along entire Batch Import module. In #154 `ComplexBatchImportComponent::prepareProjectSelect` utilize same method

Reply:
I know, this looks like a code duplication and IDE underlines it. But when we get a closer look to this code we could see, that there is just creation of an object and adding one property (`ValueChangedListener`) to it, which contains component-specific set of instructions how to deal with variable inside component that just was changed. Putting this all stuff to the `CommonComponentsFactory` would make it less _common_.

Reply:
I get it, but behavior like this - project change? save it and update datasets - is common in Batch Module, so maybe we should move it to utli class like `CommonComponentsFactory`